torch/_inductor/fx_passes/joint_graph.py

```diff
@@ -112,6 +112,65 @@ def replace_no_op(node, replace_input_index):
             replace_no_op(node, 0)
 
 
+@torch.utils._python_dispatch._disable_current_modes()
+def remove_redundant_complex_views(gm: torch.fx.GraphModule):
+    """
+    Removes redundant complex views by reusing existing ones.
+    """
+
+    # A dictionary mapping a tensor to all aliased views.
+    aliases = {}
+    graph = gm.graph
+
+    for node in graph.nodes:
+        if node.op != "call_function":
+            continue
+
+        if node.target != torch.ops.aten.view.dtype:
+            continue
+
+        src = node.args[0]
+        from_type = src.meta["val"].dtype
+        to_type = node.args[1]
+
+        # Only handle complex views.
+        if not from_type.is_complex and not to_type.is_complex:
+            continue
+
+        existing_aliases = aliases.get(src)
+        is_needed = True
+
+        if existing_aliases:
+            # Replace the view with the an existing view if available.
+            for alias in existing_aliases:
```

Comment:
This loop wouldn't be needed if you made aliases map from `(input_node, dtype) -> view_node`

Reply:
`existing_aliases` is a closure of all aliases with different types. For now since we are only dealing with complex views, the way you pointed out works. But thinking beyond complex views, without the loop we may need recursively search `aliases` for patterns like 

```
i1 = view(c1, int64)
f2 = view(i1, float32)
...
c2 = view(f2, complex64)
```
where `c1` can be replaced by `c2`.

For now with complex adds only, `existing_aliases` should only contain one float32 and one complex64.